crypto_kem/ledakemlt12/leaktime/dfr_test.c

```diff
@@ -2,70 +2,45 @@
 #include "dfr_test.h"
 #include "gf2x_arith_mod_xPplusOne.h"
 #include "qc_ldpc_parameters.h"
+#include "sort.h"
 
 #include <string.h>
 
-/* Tests if the current code attains the desired DFR. If that is the case,
- * computes the threshold for the second iteration of the decoder and returns this values
- * (max DV * M), on failure it returns 255 >> DV * M */
 
-uint8_t PQCLEAN_LEDAKEMLT12_LEAKTIME_DFR_test(POSITION_T LSparse[N0][DV * M]) {
+int PQCLEAN_LEDAKEMLT12_LEAKTIME_DFR_test(POSITION_T LSparse[N0][DV * M], uint8_t *secondIterThreshold) {
 
-    POSITION_T LSparse_loc[N0][DV * M];
-    POSITION_T rotated_column[DV * M];
-    /* Gamma matrix: an N0 x N0 block circulant matrix with block size p
-     * gamma[a][b][c] stores the intersection of the first column of the a-th
-     * block of L  with the c-th column of the b-th block of L.
-     * Gamma computation can be accelerated employing symmetry and QC properties */
+    POSITION_T LSparse_loc[N0][DV * M]; /* vector of N_0 sparse blocks */
     unsigned int gamma[N0][N0][P] = {{{0}}};
-    unsigned int gammaHist[N0][DV * M + 1] = {{0}};
     unsigned int maxMut[N0], maxMutMinusOne[N0];
-    unsigned int firstidx, secondidx, intersectionval;
     unsigned int allBlockMaxSumst, allBlockMaxSumstMinusOne;
-    unsigned int toAdd, histIdx;
+    unsigned int gammaHist[N0][DV * M + 1] = {{0}};
 
-    /*transpose blocks of L, we need its columns */
     for (int i = 0; i < N0; i++) {
         for (int j = 0; j < DV * M; j++) {
             if (LSparse[i][j] != 0) {
-                LSparse_loc[i][j] = (P - LSparse[i][j]);
+                LSparse_loc[i][j] = (P - LSparse[i][j]) ;
             }
         }
-        PQCLEAN_LEDAKEMLT12_LEAKTIME_quicksort_sparse(LSparse_loc[i]);
+        PQCLEAN_LEDAKEMLT12_LEAKTIME_uint32_sort(LSparse_loc[i], DV * M);
     }
 
+
     for (int i = 0; i < N0; i++ ) {
```

Comment:
```suggestion
     for (size_t i = 0; i < N0; i++ ) {
```

Reply:
This can probably be done in many places though, not all visible in this diff...